server/cromwell.py

```diff
@@ -0,0 +1,247 @@
+                    cwd:
+                      type: string
+                      description: to set the working directory, relative to the repo root
+                    description:
+                      type: string
+                      description: Description of the workflow to run
+                    workflow:
+                      type: string
+                      description: the relative path of the workflow (from the cwd)
+                    input_json_paths:
+                      type: List[string]
+                      description: the relative path to an inputs.json (from the cwd). Currently only supports one inputs.json
+                    dependencies:
+                      type: array
+                      items: string
+                      description: 'An array of directories (/ files) to zip for the '-p / --tools' input to "search for workflow imports"
+                    wait:
+                      type: boolean
+                      description: 'Wait for workflow to complete before returning, could yield long response times'
+
+
+          responses:
+            '200':
+              content:
+                text/plain:
+                  schema:
+                    type: string
+                    example: 'batch.hail.populationgenomics.org.au/batches/{batch}'
+              description: URL of submitted hail batch workflow
+        """
+        email = get_email_from_request(request)
+        # When accessing a missing entry in the params dict, the resulting KeyError
+        # exception gets translated to a Bad Request error in the try block below.
+        params = await request.json()
+        try:
+            output_dir = validate_output_dir(params['output'])
+            dataset = params['dataset']
+            check_dataset_and_group(dataset, email)
+            repo = params['repo']
+            check_allowed_repos(dataset, repo)
+            access_level = params['accessLevel']
+
+            ds_config = server_config[dataset]
+            project = ds_config.get('projectId')
+            hail_token = ds_config.get(f'{access_level}Token')
+            service_account_json = get_cromwell_key(
+                dataset=dataset, access_level=access_level
+            )
+            # use the email specified by the service_account_json again
+            service_account_dict = json.loads(service_account_json)
+            if 'client_email' not in service_account_dict:
+                raise web.HTTPServerError(
+                    reason="The service_account didn't contain an entry for client_email"
+                )
+            # service_account_email = f'cromwell-{access_level}@{project}.iam.gserviceaccount.com'
+            # service_account_email = f'cromwell-runner@cromwell-305305.iam.gserviceaccount.com'
+            service_account_email = service_account_dict['client_email']
+            intermediate_dir = f'gs://cpg-{dataset}-{access_level}-tmp/cromwell'
```

Comment:
This won't work for `access_level == "full"`.

Reply:
Addressed with:

```py
if access_level == 'test':
    intermediate_dir = f'gs://cpg-{dataset}-test-tmp/cromwell'
else:
    intermediate_dir = f'gs://cpg-{dataset}-main-tmp/cromwell'
```